docs/design/StratisSoftwareDesign.lyx

```diff
@@ -4944,1458 +4603,276 @@ See libdm/libdm-common.c _is_whitelisted_char() in the lvm2 code for more.
 \begin_layout Standard
-TBD
-\end_layout
-
-\begin_layout Subsubsection*
-Layer 2: Redundancy
+Since we want to allow Stratis to be used for system files, Stratis needs
+ to run in the initrd preboot environment.
+ This allows it to activate pools and filesystems so that they can be mounted
+ and accessible during the transition to the main phase of operation.
 \end_layout
 
 \begin_layout Standard
-TBD
+The use of D-Bus is not possible in the preboot environment.
+ Therefore, Stratis has an alternate IPC mechanism to be used in the initramfs.
 \end_layout
 
-\begin_layout Subsubsection*
-Layer 3: Flex
+\begin_layout Subsection
+OS Integration: udev
 \end_layout
 
 \begin_layout Standard
-\begin_inset Tabular
-<lyxtabular version="3" rows="4" columns="5">
-<features tabularvalignment="middle">
-<column alignment="center" valignment="top">
-<column alignment="center" valignment="top">
-<column alignment="center" valignment="top">
-<column alignment="center" valignment="top">
-<column alignment="center" valignment="top">
-<row>
-<cell alignment="center" valignment="top" topline="true" bottomline="true" leftline="true" usebox="none">
-\begin_inset Text
+The udev library 
+\begin_inset Quotes eld
+\end_inset
 
-\begin_layout Plain Layout
-Private
+libudev
+\begin_inset Quotes erd
+\end_inset
+
+ enables access to the udev device database.
+ This allows library users to enumerate block devices on the system, and
+ includes attributes describing their contents, such as what filesystem
+ or volume manager signature was detected.
+ (libudev uses libblkid for this, which recently had Stratis signature support
+ added.) The primary benefit of this is to perform the time-consuming block
+ device scan only once, and to alleviate library users from interpreting
+ block device contents.
 \end_layout
 
-\end_inset
-</cell>
-<cell alignment="center" valignment="top" topline="true" bottomline="true" leftline="true" usebox="none">
-\begin_inset Text
+\begin_layout Standard
+On boot, Stratis uses libudev to enumerate Stratis block devices on the
+ system, reads the Stratis metadata from each, and activates pools that
+ are complete.
+ Later, during the main running phase, Stratis monitors udev events for
+ newly-added block devices, so that if missing Stratis pool members are
+ connected to complete a pool, the pool can be activated and used.
+\end_layout
 
-\begin_layout Plain Layout
-layer-name
+\begin_layout Subsection
+OS Integration: /dev entries
 \end_layout
 
-\end_inset
-</cell>
-<cell alignment="center" valignment="top" topline="true" bottomline="true" leftline="true" usebox="none">
-\begin_inset Text
+\begin_layout Standard
+Stratis allows the user to create filesystems, which then can be mounted
+ and used via mount(8) and the fstab(5).
+ Stratisd provides a udev rule that generates a /dev/stratis directory.
+ It creates /dev/stratis/<poolname> for each pool present on the system,
+ and /dev/stratis/<poolname>/<filesystemname> for each filesystem within
```

Comment:
The paths format a bit wierd. You might want to try math environment and see if that improves things. Not a big deal, though.